sdk/test-utils/recorder-new/test/testProxyTests.spec.ts

```diff
@@ -1,17 +1,16 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-import { env, isLiveMode, isPlaybackMode } from "@azure-tools/test-recorder";
 import {
   createPipelineRequest,
   HttpMethods,
   PipelineRequestOptions
 } from "@azure/core-rest-pipeline";
 import { ServiceClient } from "@azure/core-client";
-import { TestProxyHttpClient } from "../src";
+import { env, isLiveMode, isPlaybackMode, Recorder } from "../src";
 import { expect } from "chai";
 
-type TestMode = "record" | "playback" | "live" | undefined;
+type TestMode = "record" | "playback" | "live" | "undefined";
```

Comment:
why this change?

Reply:
made this
![image](https://user-images.githubusercontent.com/10452642/146832537-5653e6d4-4514-424a-a712-5e7b6acb3916.png)

Should I do `Record<string, undefined>`?
It'll be extensive and have to check for the variables' existence everywhere.

Reply:
Strictly, `Record<string, string>` isn't an accurate representation of the underlying type of `process.env`. It might lead to bugs down the line if we assume that it is of that type (since `undefined` could crop up at runtime). Could do something like

```ts

if(!("TEST_MODE" in process.env)) {
  throw new RecorderError("TEST_MODE not defined..."); // or define TEST_MODE to default or something 🤷
}

export const env: {
  TEST_MODE: string; // and repeat for the env vars you want to guarantee exist
  [opt: string]?: string; // catchall for the other environment variables
} = process.env;
```

Reply:
> export const env: {
  TEST_MODE: string; // and repeat for the env vars you want to guarantee exist
  [opt: string]?: string; // catchall for the other environment variables
} = process.env;

TEST_MODE can also be undefined, will be considered playback.

Reply:
Well, then I'd say `Record<string, string>` is definitely not the right type. I don't know what the consequences of changing to `Record<string, string | undefined>` are but it would definitely be a step in the right direction from a correctness perspective (and it forces us to provide defaults when things might not be defined)... but if it's a huge amount of work I don't feel that strongly about it

Reply:
moved to Record<string, string | undefined>